main/ejml-dsparse/test/org/ejml/sparse/csc/misc/TestImplCommonOpsWithSemiRing_DSCC.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2009-2020, Peter Abeles. All Rights Reserved.
+ *
+ * This file is part of Efficient Java Matrix Library (EJML).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ejml.sparse.csc.misc;
+
+import org.ejml.data.DMatrixSparseCSC;
+import org.ejml.ops.DSemiRing;
+import org.ejml.ops.DSemiRings;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class TestImplCommonOpsWithSemiRing_DSCC {
+
+    @ParameterizedTest
+    @MethodSource("elementWiseAddSource")
+    public void add(DSemiRing semiRing, double[] expected) {
+        // == graph unions
+        DMatrixSparseCSC a = new DMatrixSparseCSC(3, 3);
+        DMatrixSparseCSC b = a.copy();
+        DMatrixSparseCSC c = a.copy();
+
+        a.set(1, 1, 2);
+
+        b.set(1, 1, 3);
+        b.set(0, 0, 4);
+
+        ImplCommonOpsWithSemiRing_DSCC.add(1, a, 1, b, c, semiRing, null, null);
+
+        double[] found = new double[]{c.get(0, 0), c.get(1, 1)};
+        assertTrue(c.getNumElements() == 2);
+
+        assertTrue(Arrays.equals(expected, found));
+    }
+
+    @ParameterizedTest
+    @MethodSource("elementWiseMultSource")
+    public void testelementWiseMult(DSemiRing semiRing, double[] expected) {
+        // == graph intersection
+        DMatrixSparseCSC matrix = new DMatrixSparseCSC(3, 3, 4);
+        matrix.set(1, 1, 4);
+        matrix.set(1, 2, -2);
+
+        DMatrixSparseCSC otherMatrix = matrix.copy();
+        otherMatrix.set(1, 1, 3);
+        otherMatrix.set(1, 2, 1);
+
```

Comment:
a complete nit, but these extra spaces annoy me. I do it all the time too, then I spend the next few years staring at the code resisting the urge to fix it unless I just happen to need to edit the function for some other reason. I rarely modify code for style reasons because it makes debugging regressions much more difficult.

Reply:
I removed the extra line :)